models/User.js

```diff
@@ -0,0 +1,56 @@
+const { Model, DataTypes } = require("sequelize");
+const bcrypt = require("bcrypt");
+const sequelize = require("../config/connection");
+const { Post } = require(".");
+
+class User extends Model {
+  // checkPassword is a method(loginPw is a parameter, could be anything)
+  checkPassword(loginPw) {
+    return bcrypt.compareSync(loginPw, this.password);
+  }
+}
+
+User.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      primaryKey: true,
+      autoIncrement: true,
+    },
+    username: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      unique: true,
+      validate: {
+        len: [6],
+      },
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notEmpty: false,
+        len: {
+          args: [6, 20],
+          msg: "Please provide password with 6 to 20 characters.",
```

Comment:
ensure that you either check for this in the FE js or in the controller which handles sign ups, otherwise it could potentially break your application.